tests/bson/bson-fromPHP_error-008.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+BSON\fromPHP(): PHP documents with circular references
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+echo "\nTesting packed array with circular reference\n";
+
+echo throws(function() {
+    $document = ['x' => 1, 'y' => [1, 2, 3]];
+    $document['y'][] = &$document['y'];
+    fromPHP($document);
+}, 'MongoDB\Driver\Exception\UnexpectedValueException'), "\n";
+
+echo "\nTesting associative array with circular reference\n";
+
+echo throws(function() {
+    $document = ['x' => 1, 'y' => []];
+    $document['y'][0]['z'] = &$document['y'];
+    fromPHP($document);
+}, 'MongoDB\Driver\Exception\UnexpectedValueException'), "\n";
```

Comment:
This appears to be the only new test, which is only checking that ".0." is incorporated into the exception message's field path. I think this should be be added to the `004` test as an additional case since it's so closely related to the other cases. The test case description should then be: "Testing associative array in packed array with circular reference" or something to that effect.

Noting the historic argument of not modifying existing test cases, I'd argue that spreading closely related tests out across multiple files only complicates maintainability. This would also only be a test case addition to an existing file (not altering existing tests in any way).